neo/Implementations/Notifications/LevelDB/NotificationDB.py

```diff
@@ -166,3 +210,56 @@ def get_by_addr(self, address):
                 except Exception as e:
                     logger.info("could not parse event: %s " % val)
         return results
+
+    def get_by_contract(self, contract_hash):
+        hash = contract_hash
+        if isinstance(contract_hash, str) and len(contract_hash) == 40:
+            hash = UInt160.ParseString(contract_hash)
+
+        if not isinstance(hash, UInt160):
+            raise Exception("Incorrect address format")
+
+        contractlist_snapshot = self.db.prefixed_db(NotificationPrefix.PREFIX_CONTRACT).snapshot()
+        results = []
+
+        for val in contractlist_snapshot.iterator(prefix=bytes(hash.Data), include_key=False):
+            if len(val) > 4:
+                try:
+                    event = SmartContractEvent.FromByteArray(val)
+                    results.append(event)
+                except Exception as e:
+                    logger.info("could not parse event: %s " % val)
```

Comment:
is this an info or should it show up red like an error? Then use `logger.error` or `logger.exception`